### PR TITLE
Fix PAPI command line option documentation

### LINKED ISSUE #4123: PAPI performance counters are inaccessible
## Expected Behavior
I have built HPX with support for the PAPI performance counters. The resulting collection of HPX libraries includes `libhpx_papi_counters.so`.

According to the documentation:

> For a full list of available PAPI events and their (short) description
> use the --hpx:list-counters and --papi-event-info=all command line options.

When using these options, I expect to see a list of performance counters including the names of PAPI performance counters.

## Actual Behavior
The latter option is not recognized though:

```
my_exe --hpx:list-counters --papi-event-info=all > out
runtime_support::load_components: command line processing: unrecognised
option '--papi-event-info=all'
```

A list of performance counters is printed to the screen, but does not include the names of PAPI performance counters.

## Steps to Reproduce the Problem
  1. I use CMake's FetchContent module to incorporate HPX in my project and I 
don't perform an install step. My project's targets and HPX's targets 
are put in the same bin and lib build directories. See below for the HPX configuration.
  1. Pick an HPX exe and ask for the performance counters, including the PAPI ones

## Specifications
```
{config}:
  HPX_WITH_AGAS_DUMP_REFCNT_ENTRIES=OFF
  HPX_WITH_APEX=OFF
  HPX_WITH_ATTACH_DEBUGGER_ON_TEST_FAILURE=OFF
  HPX_WITH_AUTOMATIC_SERIALIZATION_REGISTRATION=ON
  HPX_WITH_CXX14_RETURN_TYPE_DEDUCTION=TRUE
  HPX_WITH_DEPRECATION_WARNINGS=ON
  HPX_WITH_GOOGLE_PERFTOOLS=ON
  HPX_WITH_HPXMP=OFF
  HPX_WITH_IO_COUNTERS=ON
  HPX_WITH_IO_POOL=ON
  HPX_WITH_ITTNOTIFY=OFF
  HPX_WITH_LOGGING=ON
  HPX_WITH_MORE_THAN_64_THREADS=ON
  HPX_WITH_NATIVE_TLS=ON
  HPX_WITH_NETWORKING=ON
  HPX_WITH_PAPI=ON
  HPX_WITH_PARCELPORT_ACTION_COUNTERS=ON
  HPX_WITH_PARCELPORT_LIBFABRIC=OFF
  HPX_WITH_PARCELPORT_MPI=ON
  HPX_WITH_PARCELPORT_MPI_MULTITHREADED=ON
  HPX_WITH_PARCELPORT_TCP=ON
  HPX_WITH_PARCELPORT_VERBS=OFF
  HPX_WITH_PARCEL_COALESCING=ON
  HPX_WITH_PARCEL_PROFILING=OFF
  HPX_WITH_SANITIZERS=OFF
  HPX_WITH_SCHEDULER_LOCAL_STORAGE=OFF
  HPX_WITH_SPINLOCK_DEADLOCK_DETECTION=OFF
  HPX_WITH_STACKTRACES=ON
  HPX_WITH_SWAP_CONTEXT_EMULATION=OFF
  HPX_WITH_TESTS_DEBUG_LOG=OFF
  HPX_WITH_THREAD_BACKTRACE_ON_SUSPENSION=OFF
  HPX_WITH_THREAD_CREATION_AND_CLEANUP_RATES=OFF
  HPX_WITH_THREAD_CUMULATIVE_COUNTS=ON
  HPX_WITH_THREAD_DEBUG_INFO=OFF
  HPX_WITH_THREAD_DESCRIPTION_FULL=OFF
  HPX_WITH_THREAD_GUARD_PAGE=ON
  HPX_WITH_THREAD_IDLE_RATES=ON
  HPX_WITH_THREAD_LOCAL_STORAGE=OFF
  HPX_WITH_THREAD_MANAGER_IDLE_BACKOFF=ON
  HPX_WITH_THREAD_QUEUE_WAITTIME=OFF
  HPX_WITH_THREAD_STACK_MMAP=ON
  HPX_WITH_THREAD_STEALING_COUNTS=OFF
  HPX_WITH_THREAD_TARGET_ADDRESS=OFF
  HPX_WITH_TIMER_POOL=ON
  HPX_WITH_TUPLE_RVALUE_SWAP=ON
  HPX_WITH_VALGRIND=OFF
  HPX_WITH_VERIFY_LOCKS=OFF
  HPX_WITH_VERIFY_LOCKS_BACKTRACE=OFF
  HPX_WITH_VERIFY_LOCKS_GLOBALLY=OFF

  HPX_PARCEL_MAX_CONNECTIONS=512
  HPX_PARCEL_MAX_CONNECTIONS_PER_LOCALITY=4
  HPX_AGAS_LOCAL_CACHE_SIZE=4096
  HPX_HAVE_MALLOC=tcmalloc
  HPX_PREFIX (configured)=
  HPX_PREFIX=/quanta1/home/jong0137/development/objects/RelWithDebInfo/lue
{version}: V1.3.0 (AGAS: V3.0), Git: a943fd2c5d
{boost}: V1.65.1
{build-type}: release
{date}: Sep 24 2019 15:35:21
{platform}: linux
{compiler}: GNU C++ version 7.2.0
{stdlib}: GNU libstdc++ version 20170814
```

### PR DESCRIPTION
**Pull request by msimberg**
Fixes #4123.

- Fixes command line option documentation.
- Fixes loop condition for handling unregistered options.

### CODE REVIEW

File: src/runtime/components/server/runtime_support_server.cpp [hunk 1]
```diff
@@ -1089,7 +1089,7 @@ namespace hpx { namespace components { namespace server
                     &still_unregistered_options);
 
                 std::string still_unknown_commandline;
-                for (std::size_t i = 1; i != still_unregistered_options.size();
+                for (std::size_t i = 1; i < still_unregistered_options.size();
```
Comment (msimberg) [inline src/runtime/components/server/runtime_support_server.cpp]: @hkaiser would you happen to know the significance of starting at `i = 1`? It's not really clear from the context.

Comment (hkaiser) [reply src/runtime/components/server/runtime_support_server.cpp]: @msimberg: honestly, I don't remember. I'll try looking at it at runtime later.

Comment (hkaiser) [reply src/runtime/components/server/runtime_support_server.cpp]: @msimberg the first element of that vector is the equivalent to `argv[0]`, i.e. it refers to the path of the launched executable.

Comment (msimberg) [reply src/runtime/components/server/runtime_support_server.cpp]: It seems like that's the case when `allow_unknown=1`, but when it's `0` the vector can be empty. Whether that's a bug or not I can't tell.

Comment (hkaiser) [reply src/runtime/components/server/runtime_support_server.cpp]: @msimberg that could be the case and your fix addresses this issue.

**Pull request merged**